src/services/staking/staking-rewards.service.ts

```diff
@@ -66,15 +71,18 @@ export class StakingRewardsService {
     return supplies;
   }
 
-  private async getRelativeWeightsForGauges(
-    gaugeAddresses: string[],
-    customTimestamp?: number
-  ) {
-    const timestamp = customTimestamp || getUnixTime(new Date());
-    const result = await this.gaugeController.getRelativeWeights(
-      gaugeAddresses,
-      timestamp
-    );
+  private async getRelativeWeightsForGauges(gaugeAddresses: string[]) {
+    const timestamp = getUnixTime(new Date());
+    // the ve bal helpers contract for gauge weights calls
+    // the checkpoint function which is necesary for returning
+    // the correct value.
+    let result = await this.veBALHelpers.getRelativeWeights(gaugeAddresses);
+    if (configService.network.chainId === Network.KOVAN) {
```

Comment:
Rather than re-using result and doing two calls on Kovan you could move this if statement up and put a return inside it.

Reply:
done